astropy/io/misc/asdf/tags/coordinates/tests/test_earthlocation.py

```diff
@@ -49,10 +49,12 @@ def test_earthlocation_geodetic(position, ellipsoid, tmpdir):
     assert_roundtrip_tree(tree, tmpdir)
 
 
-@pytest.mark.remote_data
 def test_earthlocation_site(tmpdir):
-
-    keck = EarthLocation.of_site('Keck Observatory')
-
-    tree = dict(location=keck)
-    assert_roundtrip_tree(tree, tmpdir)
+    orig_sites = getattr(EarthLocation, '_site_registry', None)
+    try:
+        EarthLocation._get_site_registry(force_builtin=True)
+        rog = EarthLocation.of_site('greenwich')
+        tree = dict(location=rog)
+        assert_roundtrip_tree(tree, tmpdir)
+    finally:
+        EarthLocation._site_registry = orig_sites
```

Comment:
Maybe we should just make this a context manager as it seems like a useful thing to have?

Reply:
Possibly just as a `setup_class`, `teardown_class`, though for now this would seem good enough.

Reply:
I briefly thought about turning it into a test class but decided that the line count would blow up and maybe not worth it. 🤷 

I thought I opened an issue a long time ago to request changes to API allowing user to specify their own local `sites.json` but I cannot find that anymore. I think changing the API to context manager is out of scope here.